core-models/src/main/scala/com/mesosphere/usi/core/models/RunTemplateLike.scala

```diff
@@ -0,0 +1,134 @@
+package com.mesosphere.usi.core.models
+
+import com.mesosphere.usi.core.models.resources.ResourceRequirement
+import com.typesafe.scalalogging.StrictLogging
+import org.apache.mesos.v1.{Protos => Mesos}
+
+import scala.collection.JavaConverters._
+
+sealed trait RunTemplateLike {
+  final lazy val allResources: Seq[ResourceRequirement] = {
+    val b = Seq.newBuilder[ResourceRequirement]
+    taskResourceRequirements.values.foreach { r =>
+      r.foreach(b += _)
+    }
+    executorResources.foreach { b += _ }
+    b.result()
+  }
+  val taskResourceRequirements: Map[PartialTaskId, Seq[ResourceRequirement]]
+  val executorResources: Seq[ResourceRequirement]
+  val role: String
+  private[usi] def buildOperation(
+      matchedOffer: Mesos.Offer,
+      taskIdStrategy: CurriedPodTaskIdStrategy,
+      executorResources: Seq[Mesos.Resource],
+      taskResources: Map[PartialTaskId, Seq[Mesos.Resource]])
+    : Either[Mesos.Offer.Operation.Launch, Mesos.Offer.Operation.LaunchGroup]
+}
+
+class LegacyLaunchBuilder(val role: String, taskBuilder: TaskBuilderLike) extends RunTemplateLike {
+  final override val taskResourceRequirements: Map[PartialTaskId, Seq[ResourceRequirement]] =
+    Map(PartialTaskId.empty -> taskBuilder.resourceRequirements)
+
+  final override val executorResources: Seq[ResourceRequirement] = Nil
+
+  override private[usi] def buildOperation(
+      offer: Mesos.Offer,
+      taskIdStrategy: CurriedPodTaskIdStrategy,
+      executorResources: Seq[Mesos.Resource],
+      taskResources: Map[PartialTaskId, Seq[Mesos.Resource]]
+  ): Either[Mesos.Offer.Operation.Launch, Mesos.Offer.Operation.LaunchGroup] = {
+    Left(???)
+  }
+}
+
+class LaunchGroupBuilder(val role: String, executorBuilder: ExecutorBuilderLike, tasks: Map[PartialTaskId, TaskBuilderLike])
+    extends RunTemplateLike
+    with StrictLogging {
+  val taskResourceRequirements = tasks.map { case (taskId, taskBuilder) => taskId -> taskBuilder.resourceRequirements }
+
+  override val executorResources: Seq[ResourceRequirement] = Nil
+
+  override private[usi] def buildOperation(
+      matchedOffer: Mesos.Offer,
+      taskIdStrategy: CurriedPodTaskIdStrategy,
```

Comment:
If I understand correctly the user can define how tasks are named. Should we allow this flexibility?